crates/subspace-runtime/src/lib.rs

```diff
@@ -689,6 +691,47 @@ pub type Address = sp_runtime::MultiAddress<AccountId, ()>;
 pub type Header = generic::Header<BlockNumber, BlakeTwo256>;
 /// Block type as expected by this runtime.
 pub type Block = generic::Block<Header, UncheckedExtrinsic>;
+
+/// Controls non-root access to feeds and object store
+#[derive(Debug, Encode, Decode, Clone, Eq, PartialEq, Default, TypeInfo)]
+pub struct ControlNonRootStorageAccess;
```

Comment:
According to the naming convention in SignedExtra, `CheckStorageAccess` might be a better one.